samples/ClassroomLabs/Module/Library/Az.AzureLabs.psm1

```diff
@@ -0,0 +1,1282 @@
+  [CmdletBinding(SupportsShouldProcess=$false)]
+  [OutputType([System.TimeSpan])]
+
+  param(
+      [Parameter(ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true, Mandatory=$true)]
+      [Alias('ISO8601', 'String')]
+      [string]$Duration
+  )
+
+  $pattern = '^P?T?((?<Years>\d+)Y)?((?<Months>\d+)M)?((?<Weeks>\d+)W)?((?<Days>\d+)D)?(T((?<Hours>\d+)H)?((?<Minutes>\d+)M)?((?<Seconds>\d*(\.)?\d*)S)?)$'
+
+  if($Duration -match $pattern) {
+      Set-StrictMode -Off
+      $dt = [datetime]::MinValue
+      Write-Verbose (PrintHashtable -hash $Matches)
+
+      if ($Matches.Seconds) { $dt = $dt.AddSeconds($Matches.Seconds) }
+      if ($Matches.Minutes) { $dt = $dt.AddMinutes($Matches.Minutes) }
+      if ($Matches.Hours)   { $dt = $dt.AddHours($Matches.Hours) }
+      if ($Matches.Days)    { $dt = $dt.AddDays($Matches.Days) }
+      if ($Matches.Weeks)   { $dt = $dt.AddDays(7*$Matches.Weeks) }
+      if ($Matches.Months)  { $dt = $dt.AddMonths($Matches.Months) }
+      if ($Matches.Years)   { $dt = $dt.AddYears($Matches.Years) }
+      $dt - [datetime]::MinValue
+  } else {
+      Write-Warning 'The provided string does not match the ISO 8601 duration format'
+  }
+}
+
+function Get-AzureRmCachedAccessToken()
+{
+  $ErrorActionPreference = 'Stop'
+  Set-StrictMode -Off
+
+    $azureRmProfile = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureRmProfileProvider]::Instance.Profile
+    if(-not $azureRmProfile.Accounts.Count) {
+        Write-Error "Ensure you have logged in before calling this function."
+    }
+
+  $currentAzureContext = Get-AzureRmContext
+  $profileClient = New-Object Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient($azureRmProfile)
+  Write-Debug ("Getting access token for tenant" + $currentAzureContext.Subscription.TenantId)
+  $token = $profileClient.AcquireAccessToken($currentAzureContext.Subscription.TenantId)
+  $token.AccessToken
+}
+
+function GetHeaderWithAuthToken {
+
+  $authToken = Get-AzureRmCachedAccessToken
+  Write-Debug $authToken
+
+  $header = @{
+      'Content-Type' = 'application/json'
+      "Authorization" = "Bearer " + $authToken
+      "Accept" = "application/json;odata=fullmetadata"
+  }
+
+  return $header
+}
+
+$ApiVersion     = 'api-version=2019-01-01-preview'
+
+function GetLabAccountUri($ResourceGroupName) {
+    $subscriptionId = (Get-AzureRmContext).Subscription.Id
+    "https://management.azure.com/subscriptions/$subscriptionId/resourceGroups/$ResourceGroupName/providers/Microsoft.LabServices/labaccounts"
+}
+
+function ConvertToUri($resource) {
+    "https://management.azure.com" + $resource.Id
+}
+
+function InvokeRest($Uri, $Method, $Body, $params) {
+    $authHeaders = GetHeaderWithAuthToken
+
+    $fullUri = $Uri + '?' + $ApiVersion
+
+    if($params) { $fullUri += '&' + $params }
+
+    if($body) { Write-Verbose $body}    
+    $result = Invoke-WebRequest -Uri $FullUri -Method $Method -Headers $authHeaders -Body $Body -SkipHeaderValidation
```

Comment:
I just want to make sure this is setting `Content-Type` header to `application/json` when appropriate?

Reply:
I haven't found a case where it does not.